### PR TITLE
Adding e2e tests

### PR DESCRIPTION
**Pull request by bcaton85**
Revises e2e tests to follow [Cypress e2e testing guidelines](https://docs.cypress.io/guides/guides/network-requests).  

- CI now runs an instance of Quay to run tests against
- Quay test instance is seeded with data from the `/cypress/test` directory
- NPM scripts `quay:dump` and `quay:seed` have been added to store and load test data into a local instance of Quay
- Tests have been updated to use the local instance of Quay or stubbed Cypress responses

Previously tests were using the Mock API but resulted in the following issues:
- Does not guarantee correct responses from the server
- Tests were no longer isolated, updating mock data can break seemingly unrelated tests
- Have to reproduce write behavior eg. deletion and creation of repositories

### CODE REVIEW

File: cypress/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Updating the Quay seed data
+
+Both the database and blob data must be stored to seed the Quay instance. This data is stored in the `/cypress/test` directory. Running `npm run quay:seed` will seed the local instance of Quay with the test data. 
+
+To make changes to the test data:
+- Have a local instance of Quay running via `docker-compose` (`make local-dev-up`)
+- Run `npm run quay:seed` to populate the instance with the test data
+- Make required changes to the Quay instance
+- Run `npm run quay:dump` to update the `/cypress/test` directory with the test data
```
Comment (HammerMeetNail) [inline cypress/README.md:43]: Do we need to commit the dump? Can this be created on the fly before the tests run?

Comment (bcaton85) [reply cypress/README.md:43]: This allows the tests to be version controlled with the code. So updates to the tests are reviewed and pass at the same time as code changes.

File: package.json [hunk 2]
```diff
@@ -37,7 +37,11 @@
     "format": "prettier --config .prettierrc \"src/**/*.{ts,tsx}\" --write",
     "prepare": "husky install",
     "test:integration": "cypress run",
-    "start:integration": "serve -s dist -l 9000"
+    "start:integration": "serve -s dist -l 9000",
+    "quay:dump": "docker exec -t quay-db bash -c 'pg_dump -U quay -d quay -c' > ./cypress/test/quay-db-data.txt && docker cp quay-quay:/datastorage/registry/. ./cypress/test/quay-storage-data",
```
[the remark this answers is not in the document]
Comment (bcaton85) [reply package.json]: Yeah good idea. Added.

File: cypress/README.md (hunk 1 above)
Comment (bcaton85) [reply cypress/README.md:43]: The contents would have to be stored and loaded from somewhere regardless. 
> I was thinking that we'd only need to track whatever is seeding the db and not the contents. 
Can you expand on this more?

Comment (bcaton85) [reply cypress/README.md:43]: Discussed this offline. Storing contents was used for speed and easier modification to test data. Size has been kept low, currently half the size of the `src` directory.

**Review by HammerMeetNail — APPROVED** [review]
LGTM

**Pull request merged**